app/assets/javascripts/discourse/tests/integration/components/translation-test.gjs

```diff
@@ -0,0 +1,143 @@
+import { hash } from "@ember/helper";
+import { render } from "@ember/test-helpers";
+import { module, test } from "qunit";
+import Translation from "discourse/components/translation";
+import UserLink from "discourse/components/user-link";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import I18n from "discourse-i18n";
+
+module("Integration | Component | Translation", function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this._locale = I18n.locale;
+    this._translations = I18n.translations;
+
+    I18n.locale = "fr";
+
+    I18n.translations = {
+      fr: {
+        js: {
+          hello: "Bonjour, %{username}",
+          simple_text: "Simple text without placeholders",
+          with_options: "Hello %{name}, welcome to %{site}!",
+          multiple_placeholders:
+            "User %{user} commented on %{topic} at %{time}",
+          mixed_placeholders:
+            "Welcome %{username}! You have %{count} messages.",
```

Comment:
```suggestion
          mixed_placeholders:
            "Welcome %{username}! You have %{count} messages.",
          repeated_placeholders:
            "Welcome %{username}! Hello %{username}!
```

Repeated placeholders seem like a worthwhile case to test as well.

Reply:
Ah, I see how that can work. Thanks!